### PR TITLE
Allow to order tasks by last action in reverse order hotosm#4083 

### PR DESCRIPTION
**Pull request by max-keviv**
#4083
-Added option "least recently updated" to task ordering on task selection page
-Renamed  "Last updated first" to 'Mostly recently updated"

Screenshot from 2021-01-13 11-12-31
Screenshot from 2021-01-13 11-12-39

### CODE REVIEW

File: frontend/src/components/taskSelection/contributions.js [hunk 1]
```diff
@@ -127,10 +127,15 @@ const Contributions = ({ project, tasks, contribsData, activeUser, activeStatus,
   const { percentMapped, percentValidated, percentBadImagery } = useComputeCompleteness(tasks);
 
   let contributionsArray = contribsData.userContributions || [];
+
```
Comment (willemarcel) [inline frontend/src/components/taskSelection/contributions.js:130]: @max-keviv could you undo the changes on that file, as we are handling it in another PR?

File: frontend/src/components/taskSelection/messages.js [hunk 2]
```diff
@@ -283,9 +283,13 @@ export default defineMessages({
     id: 'project.tasks.sorting.id',
     defaultMessage: 'Sort by task number',
   },
-  sortByLastUpdate: {
-    id: 'project.tasks.sorting.date',
-    defaultMessage: 'Last updated first',
+  sortByMostRecentUpdate: {
+    id: 'project.tasks.sorting.taskdate',
```
Comment (willemarcel) [inline frontend/src/components/taskSelection/messages.js:287]: Could you modify it to `project.tasks.sorting.date`?

File: frontend/src/components/taskSelection/messages.js [hunk 3]
```diff
@@ -283,9 +283,13 @@ export default defineMessages({
...
+    defaultMessage: 'Mostly recently updated',
+  },
+  sortByLeastRecentlyUpdate: {
+    id: 'project.tasks.reverse.sorting.date',
```
Comment (willemarcel) [inline frontend/src/components/taskSelection/messages.js:291]: Could you modify it to `project.tasks.sorting.date.reverse`?

File: frontend/src/components/taskSelection/taskList.js [hunk 4]
```diff
@@ -42,8 +42,8 @@ export function TaskStatus({ status, lockHolder }: Object) {
             }}
           />
         ) : (
-          <FormattedMessage {...messages[`taskStatus_${status}`]} />
-        )}
+            <FormattedMessage {...messages[`taskStatus_${status}`]} />
```
Comment (willemarcel) [inline frontend/src/components/taskSelection/taskList.js:45]: Please fix the indentation

File: frontend/src/components/taskSelection/taskList.js [hunk 5]
```diff
@@ -269,15 +266,14 @@ export function TaskList({
...
-            }`}
+            className={`absolute w1 h1 top-0 red pt3 pointer pr3 right-0 ${textSearch ? 'dib' : 'dn'
+              }`}
           />
         </div>
         <div className="w-60-l w-50-m w-100 dib pv1">
           <Dropdown
-            onAdd={() => {}}
-            onRemove={() => {}}
+            onAdd={() => { }}
```
Comment (willemarcel) [inline frontend/src/components/taskSelection/taskList.js:275]: We don't need the blank space inside the brackets

File: frontend/src/components/taskSelection/taskList.js [hunk 6]
```diff
@@ -301,7 +297,7 @@ export function TaskList({
           <PaginatedList
             pageSize={6}
             items={
-              sortBy === 'id' ? readyTasks.sort(compareTaskId) : readyTasks.sort(compareLastUpdate)
+              sortBy === 'id' ? readyTasks.sort(compareTaskId) : sortBy === 'date' ? readyTasks.sort(compareLastUpdate) : (readyTasks.sort(compareLastUpdate)).reverse()
```
Comment (willemarcel) [inline frontend/src/components/taskSelection/taskList.js:300]: the line is too long

File: frontend/src/components/taskSelection/taskList.js [hunk 7]
```diff
@@ -354,19 +350,19 @@ function TaskActivityModal({
...
-              </h3>
-              <p className="pb0">
-                <FormattedMessage
-                  {...messages.taskSplitDescription}
-                  values={{ id: <b>#{taskId}</b> }}
-                />
-              </p>
-            </div>
-          )}
+              <div className="w-100 pa4 blue-dark bg-white">
```
Comment (willemarcel) [inline frontend/src/components/taskSelection/taskList.js:353]: Fix the indentation.

**Review by willemarcel — CHANGES_REQUESTED** [review]
Some suggestions of improvements. Most on the code style.

File: frontend/src/components/taskSelection/messages.js (hunk 2 above)
Comment (max-keviv) [reply frontend/src/components/taskSelection/messages.js:287]: @willemarcel  when i am changing it to back to 'project.tasks.sorting.date' then instead of "Mostly recently updated" "Last updated first" is showing

File: frontend/src/components/taskSelection/taskList.js (hunk 6 above)
Comment (max-keviv) [reply frontend/src/components/taskSelection/taskList.js:300]: @willemarcel  can  you suggest something else

File: frontend/src/components/taskSelection/messages.js (hunk 2 above)
Comment (willemarcel) [reply frontend/src/components/taskSelection/messages.js:287]: maybe you need to run yarn build-locales to have the updated text.

File: frontend/src/components/taskSelection/taskList.js (hunk 6 above)
Comment (willemarcel) [reply frontend/src/components/taskSelection/taskList.js:300]: you can just break the lines after the `?` and `:`

**Comment by max-keviv** [comment]
@willemarcel i am opening another pull request for this issue #4186. please review it
